pkg/client/kubernetes/client.go

```diff
@@ -185,35 +197,62 @@ func RESTConfigFromKubeconfig(kubeconfig []byte) (*rest.Config, error) {
 	return restConfig, nil
 }
 
-func validateClientConfig(clientConfig clientcmd.ClientConfig) error {
+func validateClientConfig(clientConfig clientcmd.ClientConfig, allowedFields []string) error {
+	if clientConfig == nil {
+		return nil
+	}
+
 	rawConfig, err := clientConfig.RawConfig()
 	if err != nil {
 		return err
 	}
-	return ValidateConfig(rawConfig)
+	return ValidateConfigWithAllowList(rawConfig, allowedFields)
 }
 
 // ValidateConfig validates that the auth info of a given kubeconfig doesn't have unsupported fields.
 func ValidateConfig(config clientcmdapi.Config) error {
+	return ValidateConfigWithAllowList(config, []string{})
+}
+
+// ValidateConfigWithAllowList validates that the auth info of a given kubeconfig doesn't have unsupported fields. It takes an additional list of allowed fields.
+func ValidateConfigWithAllowList(config clientcmdapi.Config, allowedFields []string) error {
 	validFields := []string{"client-certificate-data", "client-key-data", "token", "username", "password"}
+	validFields = append(validFields, allowedFields...)
 
 	for user, authInfo := range config.AuthInfos {
 		switch {
 		case authInfo.ClientCertificate != "":
+			if utils.ValueExists(AuthClientCertificate, validFields) {
```

Comment:
Nit: You could move this condition into the `case` itself to save a few LOC, e.g.

```golang
		case authInfo.ClientCertificate != "" && !utils.ValueExists(AuthClientCertificate, validFields):
```

(same for others)